### DISCUSSION TITLE
scalene show cpu time for a line

### DISCUSSION THREAD

**Discussion by bahcruz** [plasma-umass/scalene#793] [Q&A]
I am using python3.  I run 
scalene --html --cpu --profile-all --outfile sca_profile.html foo.py
But I only see profiling on functions not lines in the function.  what am I doing wrong?

**Comment by emeryberger** [comment]
Can you share a screenshot? Also, please indicate what version of Scalene you are using.

**Reply by bahcruz** [reply]
Scalene: a high-precision CPU and memory profiler, version 1.5.34 (2024.01.27)
![sca2](https://github.com/plasma-umass/scalene/assets/11378423/a3c274d1-70b0-4805-8bdb-39439b3e8cbb)

It notes in the output html, 'function summary'.   The line number listed is the declaration of the 'methodname'.  I wanted a break down of which line in 'methodname' may have caused these readings.  Not just the method overall.  How do I get more than the function summary?

**Comment by emeryberger** [comment]
Please generate a JSON file and then load it with the viewer (and attach the screenshot and the JSON file). Scalene lists individual lines by default - you don't have to ask for it - and I'd expect to see it in the viewer. There is a caveat, which is that individual lines are not listed if it doesn't account for 1% of CPU/GPU time or IIRC 10MB of allocation.

**Reply by bahcruz** [reply]
I'll work on this and get back to this post.  I appreciate your comments.

**Reply by bahcruz** [reply]
Here is the json file in the viewer (https://plasma-umass.org/scalene-gui/).  Is this the viewer you mean?  
![sca4](https://github.com/plasma-umass/scalene/assets/11378423/5c96a40f-de6e-424e-b61c-f70f2ffbebba)

I cannot attach the JSON file.  My issue is why is there nothing under 'LINE PROFILE'?

**Reply by emeryberger** [reply]
Can you mail me the JSON file? If not, you should be able to inspect it and see which lines have CPU time / GPU time / memory consumption. The amount of time as a percentage above is really low. You might need to widen your profile selection with `--profile-all` and then some `--profile-exclude` or `--profile-only` directives.

**Reply by bahcruz** [reply]
I'll look into that.  Thanks for your help.
